### PR TITLE
[PLAYER-3032] Update to CAF

### PR DESCRIPTION
**Pull request by MortMerr**

### CODE REVIEW

**Comment by marcosmx** [comment]
Hi @MortMerr LGTM, but this PR should be placed at the internal repository for the cast receiver library, let me know if you need assistance on this

File: src/app/cast.js [hunk 1]
```diff
@@ -1,64 +1,62 @@
...
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.SET_VOLUME, volumeRequestData => {
+    logger.info(LOG_PREFIX, "SET_VOLUME", volumeRequestData);
+    castPlayer.setVolume(volumeRequestData.volume);
+    return volumeRequestData;
+});
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.SEEK, seekRequestData => {
+    logger.info(LOG_PREFIX, "SEEK", seekRequestData);
+    castPlayer.seek(seekRequestData.currentTime);
+    castPlayer.onSeeked(_, seekRequestData.currentTime);
```
Comment (marcosmx) [inline src/app/cast.js]: This shouldn't be called here. The player will trigger the "seeked" event after it finished to buffer the asset at the given time.

File: src/app/cast.js [hunk 2]
```diff
@@ -1,64 +1,62 @@
...
-mediaManager.onPause = function(e) {
-    logger.info("MediaManager:onPause", e);
+const context = cast.framework.CastReceiverContext.getInstance();
+const playerManager = context.getPlayerManager();
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.LOAD, loadRequestData => {
+    logger.info(LOG_PREFIX, "LOAD", loadRequestData);
+    castPlayer.setAsset(loadRequestData.media.customData);
+    castPlayer.load();
+    playerManager.setMediaElement(castPlayer.OOPlayer);
```
Comment (marcosmx) [inline src/app/cast.js]: I feel that there is no need to set the media element every time the sender tries to load an asset on the receiver. I mean at the end the media element will be always the same. I think we need to find a way to identify the very first time the receiver creates the mediaElement and then pass it to the playerManager.

Let me know your inputs on this.

File: src/app/cast.js (hunk 1 above)
Comment (MortMerr) [reply src/app/cast.js]: Yes, it's obviously that callback should be fired by message bus. But i have a problem, that this callback doesn't fire, and i have not enough knowledge in web player api to understand why. Maybe u have some idea?

**Comment by MortMerr** [comment]
@marcograray sorry, silly mistake, i will place this pr to internal repo.

**Pull request closed without merging**